app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PagesController < ApplicationController
+  def home
+    @products = Product.all
+    
+    if params[:category] == 'entree'
+      @products = @products.where(category: 0)
+    elsif params[:category] == 'second_course'
+      @products = @products.where(category: 1)
+    elsif params[:category] == "salad"
+      @products = @products.where(category: 2)
+    elsif params[:category] == "pizza"
+      @products = @products.where(category: 3)
+    elsif params[:category] == "dessert"
+      @products = @products.where(category: 4)
+    end
+
+    if params[:vegetarian] == 'non_vegetarian'
+      @products = @products.where(vegetarian: 0)
+    elsif params[:vegetarian] == 'vegetarian'
+      @products = @products.where(vegetarian: 1)
+    end
+
+    if params[:order] == 'descending'
```

Comment:
you could send from the front-end `asc` or `desc` instead of `ascending` and `descending` and rewrite this code a bit